pkg/ddc/base/operation.go

```diff
@@ -159,39 +167,84 @@ func (t *TemplateEngine) reconcileComplete(ctx cruntime.ReconcileRequestContext,
 	opStatus *datav1alpha1.OperationStatus, operation dataoperation.OperationInterface) (ctrl.Result, error) {
 	log := ctx.Log.WithName("reconcileComplete")
 
-	// 1. Update the infos field
+	// todo when cron, need to check if excuting
+	// 1. check and update data operation's status by helm status
+	statusHandler := operation.GetStatusHandler(object)
+	if statusHandler == nil {
+		err := fmt.Errorf("fail to get status handler")
+		log.Error(err, "status handler is nil")
+		return utils.RequeueIfError(err)
+	}
+	err := statusHandler.UpdateStatusByHelmStatus(ctx, object, opStatus)
+	if err != nil {
+		log.Error(err, "failed to update status")
+		return utils.RequeueIfError(err)
+	}
+	if opStatus.Phase != common.PhaseComplete {
```

Comment:
Which scenario does the logic here correspond to?

Reply:
But if cronjob does not create another job, the cron migrate status can be `Pending` -> `Executing` -> `Completed` ->  `Pending`(CronJob Reconcile) -> `Executing` -> ... ?
is it possible when cronjob started but the Dataset lock is not locked? Should cron migrate have `Completed` status ?

Reply:
i think I don't fully understand what you're thinking. I wrote a [design doc](https://docs.google.com/document/d/1oUYBxaTNfS8VTzAHPVKAlIxp-z_k8TZHQJvL3iDV2f0/edit#) and fix some bugs. Please have a look on it, If you still have questions, we can communicate offline.